fairlearn/reductions/_exponentiated_gradient/_lagrangian.py

```diff
@@ -43,11 +49,19 @@ class _Lagrangian:
         (defaults to `sample_weight`)
     """
 
-    def __init__(self, X, y, estimator, constraints, B, opt_lambda=True,
+    def __init__(self, X, y, estimator, constraints, *, B, objective=None, opt_lambda=True,
                  sample_weight_name='sample_weight', **kwargs):
         self.constraints = constraints
         self.constraints.load_data(X, y, **kwargs)
-        self.obj = self.constraints.default_objective()
+        if objective is None:
+            self.obj = self.constraints.default_objective()
+        elif ((isinstance(constraints, ClassificationMoment)
+               and isinstance(objective, ClassificationMoment)) or
+              (isinstance(constraints, LossMoment)
+               and isinstance(objective, LossMoment))):
+            self.obj = objective
+        else:
+            raise ValueError(_MESSAGE_BAD_OBJECTIVE)
```

Comment:
Might be nice to let the users know what they supplied as the types in the error message.

Reply:
Also, I'm not seeing unit tests for this `if` block.

Reply:
[done on both accounts]